### DISCUSSION TITLE
I stuck in the triangle.exe

### DISCUSSION THREAD

**Discussion by Alvin-prc** [modflowpy/flopy#1948] [Q&A] [answered]
## I'm learning the "Triangular mesh example" in https://flopy.readthedocs.io/en/stable/Notebooks/dis_triangle_example.html.

I am not sure if I have compiled the correct exe file, anyway I put the exe file in the root directory and still get an error:

from flopy.utils.triangle import Triangle
tri = Triangle(maximum_area=500, angle=30, model_ws=workspace)
tri.add_polygon(circle_poly)
tri.build(verbose=False)
fig = plt.figure(figsize=(10, 10))
ax = plt.subplot(1, 1, 1, aspect="equal")
pc = tri.plot(ax=ax)

AttributeError                            Traceback (most recent call last)
Cell In[17], line 7
      5 tri = Triangle(maximum_area=500, angle=30, model_ws=workspace)
      6 tri.add_polygon(circle_poly)
----> 7 tri.build(verbose=False)
      8 fig = plt.figure(figsize=(10, 10))
      9 ax = plt.subplot(1, 1, 1, aspect="equal")

File C:\ProgramData\Anaconda3\lib\site-packages\flopy\utils\triangle.py:189, in Triangle.build(self, verbose)
    187 # load the results
    188 self._load_results()
--> 189 self.ncpl = self.ele.shape[0]
    190 self.nvert = self.node.shape[0]
    192 # create verts and iverts

AttributeError: 'NoneType' object has no attribute 'shape'

## Can you tell me what might be wrong or give me a correct triangle.exe? My computer is Win 11

**Comment by wpbonelli** [comment, marked as answer]
The triangle program is distributed with the rest of the [MODFLOW executables](https://github.com/MODFLOW-USGS/executables). You might try with the distributed executable instead of building triangle from source.

Your error indicates `self.ele` is `None`, which would occur if the [`.ele` output file](https://www.cs.cmu.edu/~quake/triangle.ele.html) was not produced as expected. (Perhaps flopy's triangle wrapper should warn in that case.)

**Reply by Alvin-prc** [reply]
Appreciate it

**Comment by jdhughes-dev** [comment]
triangle is also available from conda-forge
